src/test/java/com/incognia/TokenAwareDispatcher.java

```diff
@@ -0,0 +1,98 @@
+package com.incognia;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.stream.Collectors;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.SneakyThrows;
+import okhttp3.mockwebserver.Dispatcher;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.jetbrains.annotations.NotNull;
+
+public class TokenAwareDispatcher extends Dispatcher {
+  private final String token;
+  private final String clientId;
+  private final String clientSecret;
+  private final ObjectMapper objectMapper;
+  @Setter private String expectedInstallationId;
+  @Setter private String expectedAddressLine;
+  @Getter private int tokenRequestCount;
+
+  public TokenAwareDispatcher(String token, String clientId, String clientSecret) {
+    this.token = token;
+    this.clientId = clientId;
+    this.clientSecret = clientSecret;
+    this.tokenRequestCount = 0;
+    this.objectMapper = ObjectMapperFactory.OBJECT_MAPPER;
+  }
+
+  @SneakyThrows
+  @NotNull
+  @Override
+  public MockResponse dispatch(@NotNull RecordedRequest request) {
+    if ("/api/v2/onboarding".equals(request.getPath()) && "POST".equals(request.getMethod())) {
+      assertThat(request.getHeader("Content-Type")).contains("application/json");
+      assertThat(request.getHeader("Authorization")).isEqualTo("Bearer " + token);
+      return new MockResponse().setResponseCode(200).setBody("{\"name\": \"my awesome name\"}");
+    }
+    if ("/api/v2/onboarding/signups".equals(request.getPath())
+        && "POST".equals(request.getMethod())) {
+      assertThat(request.getHeader("Content-Type")).contains("application/json");
+      assertThat(request.getHeader("Authorization")).isEqualTo("Bearer " + token);
+      PostSignupRequestBody postSignupRequestBody =
+          objectMapper.readValue(request.getBody().inputStream(), PostSignupRequestBody.class);
+      assertThat(postSignupRequestBody.getInstallationId()).isEqualTo(expectedInstallationId);
+      assertThat(postSignupRequestBody.getAddressLine()).isEqualTo(expectedAddressLine);
+      String response = getResourceFileAsString("post_onboarding_response.json");
+      return new MockResponse().setResponseCode(200).setBody(response);
```

Comment:
Super Nit : This Method seems a little bit cluttered for my taste, i would probably add some empty lines in order to better organize the actions in sections of their own, like : 
```suggestion
        && "POST".equals(request.getMethod())) {
      assertThat(request.getHeader("Content-Type")).contains("application/json");
      assertThat(request.getHeader("Authorization")).isEqualTo("Bearer " + token);
      
      PostSignupRequestBody postSignupRequestBody =
          objectMapper.readValue(request.getBody().inputStream(), PostSignupRequestBody.class);
      
      assertThat(postSignupRequestBody.getInstallationId()).isEqualTo(expectedInstallationId);
      assertThat(postSignupRequestBody.getAddressLine()).isEqualTo(expectedAddressLine);
      
      String response = getResourceFileAsString("post_onboarding_response.json");
      return new MockResponse().setResponseCode(200).setBody(response);
```

But i understand this comes completely from taste :smile:

Reply:
I believe it would improve its readability if "handle" methods are extracted for each route

Reply:
yeah i believe this would be great too

Reply:
done!